astropy/tests/runner.py

```diff
@@ -178,7 +180,24 @@ def _generate_args(self, **kwargs):
         """
 
     def run_tests(self, **kwargs):
-        if not _has_test_dependencies(): # pragma: no cover
+
+        # The following option will include eggs inside a .eggs folder in
+        # sys.path when running the tests. This is possible so that when
+        # runnning python setup.py test, test dependencies installed via e.g.
+        # tests_requires are available here. This is not an advertised option
+        # since it is only for internal use
+        if kwargs.pop('add_local_eggs_to_path', False):
+
+            # Add each egg to sys.path individually
+            for egg in glob.glob(os.path.join('.eggs', '*.egg')):
+                sys.path.append(egg)
```

Comment:
Is `append` the right action here? I'm just thinking about what's going to happen if a different version of the same package is installed in the user's environment.

Reply:
Hmm yes I should append at the start